src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1187,8 +1189,13 @@ void goto_instrument_parse_optionst::instrument_goto_program()
 
   if(cmdline.isset("generate-function-body"))
   {
+    optionst options;
+    parse_c_object_factory_options(cmdline, options);
+    c_object_factory_parameterst object_factory_parameters(options);
```

Comment:
The function body generation is currently hardcoded to use the `c_nondet_symbol_factory`. I'm not sure at this point how difficult it would be to also make it work with the java object factory. I'd rather have this PR only include the C feature for now.

Reply:
Ok!